rpc/src/v1/impls/light/parity.rs

```diff
@@ -395,7 +395,7 @@ impl Parity for ParityClient {
 
 		let engine = self.light_dispatch.client.engine().clone();
 		let from_encoded = move |encoded: encoded::Header| {
-			let header = encoded.decode();
+			let header = encoded.decode().expect("decoding error"); // REVIEW: not sure what to do here; what is a decent return value for the error case here?
```

Comment:
I believe it should be `BoxFuture<Result<RichHeader>>`. 

if `self.fetcher` is fetching headers from the network, this may be a critical vulnerability in light clients